src/datatypes/src/value.rs

```diff
@@ -62,6 +63,47 @@ pub enum Value {
     List(ListValue),
 }
 
+impl Display for Value {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match *self {
```

Comment:
```suggestion
        match self {
```

Looks like this deref isn't necessary? As well as the following `ref` (e.g. `Value::List(ref v)`)

Reply:
This deref was recommended by clippy, not sure how to avoid it.
As for `ref` in `Value::List` - I used it to avoid move of value (List doesn't support Copy)

Reply:
I've made these change and clippy doesn't say anything, does it works for you?
```diff
diff --git a/src/datatypes/src/value.rs b/src/datatypes/src/value.rs
index 279a9c5..d5e0ae3 100644
--- a/src/datatypes/src/value.rs
+++ b/src/datatypes/src/value.rs
@@ -65,7 +65,7 @@ pub enum Value {
 
 impl Display for Value {
     fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
-        match *self {
+        match self {
             Value::Null => write!(f, "{}", self.data_type().name()),
             Value::Boolean(v) => write!(f, "{}", v),
             Value::UInt8(v) => write!(f, "{}", v),
@@ -78,8 +78,8 @@ impl Display for Value {
             Value::Int64(v) => write!(f, "{}", v),
             Value::Float32(v) => write!(f, "{}", v),
             Value::Float64(v) => write!(f, "{}", v),
-            Value::String(ref v) => write!(f, "{}", v.as_utf8()),
-            Value::Binary(ref v) => {
+            Value::String(v) => write!(f, "{}", v.as_utf8()),
+            Value::Binary(v) => {
                 let hex = v
                     .iter()
                     .map(|b| format!("{:02x}", b))
@@ -90,7 +90,7 @@ impl Display for Value {
             Value::Date(v) => write!(f, "{}", v),
             Value::DateTime(v) => write!(f, "{}", v),
             Value::Timestamp(v) => write!(f, "{}", v.to_iso8601_string()),
-            Value::List(ref v) => {
+            Value::List(v) => {
                 let default = Box::new(vec![]);
                 let items = v.items().as_ref().unwrap_or(&default);
                 let items = items
(END)
```

>As for ref in Value::List - I used it to avoid move of value (List doesn't support Copy)

If `self` is not deref-ed, its type is `&Value`, and the list will be `&List` correspondingly, which I think can solve this problem.

Reply:
Yes, it works, thx!
However, I don't understand why can we write `Value::Null` in branch instead of `&Value::Null` when we match on `&Value`...

Reply:
>However, I don't understand why can we write Value::Null in branch instead of &Value::Null when we match on &Value...

I guess the reason is `Value::Null` is an enum variant rather than a variable. Rust analyzer tell me the `v` in `Value::String(v)` is with type `&String`